### PR TITLE
Add Cache-Control header to static resources to help caching

### PR DESCRIPTION
**Pull request by nzmsv**
# Changes

Prior to this change putting the dashboard behind ALB broke static resource caching.  ALB seems to remove the Date header from the response it proxies, however it will obey explicit Cache-Control headers.

Static resources coming from webpack already have a content hash as part of their name so they are already immutable.

Note: the first draft of this PR used the Etag header instead.  See the discussion below.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

**Comment by a-roberts** [comment]
/ok-to-test
@AlanGreene FYI

**Comment by nzmsv** [comment]
> @nzmsv thanks for the PR, adding caching support would be very useful for a lot of use cases.
> 
> Since the URLs for the static resources already contain a content-based hash we know that content can be cached long-term without requiring periodic revalidation as in the case of using entity tags or similar approaches. Instead of sending the Etag header in the response, we could set a long max-age on the Cache-Control header and eliminate the conditional requests from the browser.
> 
> Something like `Cache-Control: public, max-age=31536000, immutable`
> 
> Would this work for your use case?

Thanks, this is a great idea!  I tested the Cache-Control header behind ALB and it works correctly.  I updated the PR accordingly.

File: pkg/router/router.go [hunk 1]
```diff
@@ -59,6 +60,7 @@ const ExtensionDisplayNameKey = "tekton-dashboard-display-name"
 const ExtensionRoot = "/v1/extensions"
 
 var webResourcesDir = os.Getenv("WEB_RESOURCES_DIR")
+var webResourcesPattern = regexp.MustCompile("^/([[:alnum:]]+\\.)?[[:alnum:]]+\\.(js)|(css)|(png)$")
```
Comment (AlanGreene) [inline pkg/router/router.go]: The favicon (the only png I see loaded) does not include a hash, so if we can exclude it for now this PR should be good to go. We'll need to make some changes on the frontend for the favicon so I'll create an issue to track that separately.

```suggestion
var webResourcesPattern = regexp.MustCompile("^/([[:alnum:]]+\\.)?[[:alnum:]]+\\.(js)|(css)$")
```

Comment (nzmsv) [reply pkg/router/router.go]: We also load the Tekton logo png (in the upper right corner).  I'll add a specific exclusion for the favicon.

**Pull request merged**